### PR TITLE
CORTX-28560: Fix multiple events publish issue (Alert publish validation skipped due to new change in schema)

### PR DESCRIPTION
**Pull request by mariyappanp**
CORTX-28560: Alert publish validation skipped due to new change in schema

# Problem Statement
- Repeating already published k8s_monitor events 

# Design
Previously Event.json returned,
{'event': { 'header': {}, 'payload': {} }}
Now,
"{ 'header': {}, 'payload': {} }"
[CORTX-28560](https://jts.seagate.com/browse/CORTX-28560)

# Coding
-  [x] Coding conventions are followed and code is consistent

# Testing 
- [x] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
```
[root@cortx-ha-headless-svc ~]# python3 test_no_duplicate_alerts.py
******** Testing K8s Not Publishing Duplicate Alerts ********
2022-03-22 08:35:05 test_k8s_resource_monitor [360]: INFO [init] Started logging for service test_k8s_resource_monitor
Producer id: mock_producer, message_type: mock_message, partition: 1
2022-03-22 08:35:05 test_k8s_resource_monitor [360]: INFO [__init__] Initialization done for pod monitor
Publishing alert..
{'header': {'version': '1.0', 'timestamp': '112255', 'event_id': '11225568919c5ba8f88ad24cc18f632a75de115aca'}, 'payload': {'source': 'monitor', 'cluster_id': 'None', 'site_id': 'None', 'rack_id': 'None', 'storageset_id': 'None', 'node_id': 'dummy.colo.seagate.com', 'resource_type': 'host', 'resource_id': 'dummy.colo.seagate.com', 'resource_status': 'online', 'specific_info': {}}}
Publishing alert..
{'header': {'version': '1.0', 'timestamp': '112233', 'event_id': '11223368911c9b81b8d3c84305a49c191e6a8c3bb5'}, 'payload': {'source': 'monitor', 'cluster_id': 'None', 'site_id': 'None', 'rack_id': 'None', 'storageset_id': 'None', 'node_id': 'dummy0fff6f947e8aecc6a27a25b7329', 'resource_type': 'node', 'resource_id': 'dummy02fff6f947e8aecc6a27a25b7329', 'resource_status': 'online', 'specific_info': {'generation_id': 'cortx-data-dummy0'}}}
Publishing alert..
{'header': {'version': '1.0', 'timestamp': '112233', 'event_id': '11223368911c9b81b8d3c84305a49c191e6a8c3bb5'}, 'payload': {'source': 'monitor', 'cluster_id': 'None', 'site_id': 'None', 'rack_id': 'None', 'storageset_id': 'None', 'node_id': 'dummy0fff6f947e8aecc6a27a25b7329', 'resource_type': 'node', 'resource_id': 'dummy02fff6f947e8aecc6a27a25b7329', 'resource_status': 'offline', 'specific_info': {'generation_id': 'cortx-data-dummy0'}}}
Successfully verified the alert.
```

# Review Checklist 
- [x] PR is self reviewed
- [x] JIRA number/GitHub Issue added to PR
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: 
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N]
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: ha/monitor/k8s/object_monitor.py [hunk 1]
```diff
@@ -187,36 +187,41 @@ def _is_published_alert(self, alert) -> bool:
...
         """
-        if not isinstance(alert, dict):
-            return False
+        if isinstance(alert, str):
+            alert = json.loads(alert)
 
         incoming_alert = copy.deepcopy(alert)
 
-        header = incoming_alert["event"]["header"]
-        payload = incoming_alert["event"]["payload"]
```
Comment (mariyappanp) [inline ha/monitor/k8s/object_monitor.py:196]: @mukhtar-inamdar 'event' is not present now in alert.

File: ha/monitor/k8s/object_monitor.py [hunk 2]
```diff
@@ -187,36 +187,41 @@ def _is_published_alert(self, alert) -> bool:
...
 
-        incoming_alert_msg = json.dumps(payload, sort_keys=True)
-        published_alert = self._published_alerts.get(alert_key)
+        except KeyError as err:
+            Log.error(f"Alert validation failed as key {err} not found")
 
-        if incoming_alert_msg == published_alert:
-            # Published already
-            return True
         else:
```
Comment (Madhura-08) [inline ha/monitor/k8s/object_monitor.py:216]: Sorry, but where is the respective "if" for this?

Comment (Madhura-08) [reply ha/monitor/k8s/object_monitor.py:216]: Please refer this comment for newly added changes of your's..!

Comment (Madhura-08) [reply ha/monitor/k8s/object_monitor.py:216]: Discussed with Mariyappan. I got it now. Thanks!

**Review by Madhura-08 — APPROVED** [review]

File: ha/monitor/k8s/object_monitor.py (hunk 1 above)
Comment (mukhtar-inamdar) [reply ha/monitor/k8s/object_monitor.py:196]: ok

**Review by mukhtar-inamdar — APPROVED** [review]

**Review by akash2144 — APPROVED** [review]

**Review by gauri-bhosale — APPROVED** [review]

File: ha/monitor/k8s/object_monitor.py [hunk 3]
```diff
@@ -187,36 +187,41 @@ def _is_published_alert(self, alert) -> bool:
             True if it is published already
             False if it is a new alert
         """
-        if not isinstance(alert, dict):
-            return False
+        if isinstance(alert, str):
```
Comment (ArchanaLimaye) [inline ha/monitor/k8s/object_monitor.py:190]: why is this change needed ?

**Comment by shaileshsvc** [comment]
Please add unit testing results (if those are in files attached in Jira, please mention that here).

File: ha/monitor/k8s/object_monitor.py (hunk 3 above)
Comment (ArchanaLimaye) [reply ha/monitor/k8s/object_monitor.py:190]: Discussion done. Understood

**Comment by mariyappanp** [comment]
> Please add unit testing results (if those are in files attached in Jira, please mention that here).

Added unittest result, @shaileshsvc

**Review by ArchanaLimaye — APPROVED** [review]

**Review by shaileshsvc — COMMENTED** [review]
Approved
